packages/@aws-cdk/aws-cognito/README.md

```diff
@@ -345,6 +355,40 @@ The following table lists the set of triggers available, and their corresponding
 For more information on the function of these triggers and how to configure them, read [User Pool Workflows with
 Triggers](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html).
 
+#### Trigger Permissions
+
+You can provide permissions to your Lambda trigger using the `attachToRolePolicy` API which will update
+the execution role used by the Lambda function.
+
+⚠️ Using the `attachToRolePolicy` API to provide permissions scoped to your user pool will result in a circular dependency.
+Error message when running `cdk synth` or `cdk deploy`:
+> Circular dependency between resources: [pool056F3F7E, fnPostAuthFnCognitoA630A2B1, ...]
+
+To work around the circular dependency issue, your Lambda trigger can be granted permissions to perform operations
+on your user pool using the `attachInlinePolicy` API.
+
+```ts
+const postAuthFn = new lambda.Function(this, 'postAuthFn', {
+  code: lambda.Code.fromInline('post authentication'),
+  runtime: lambda.Runtime.NODEJS_12_X,
+  handler: 'index.handler',
+});
+
+const userpool = new cognito.UserPool(this, 'myuserpool', {
+  lambdaTriggers: {
+    postAuthentication: postAuthFn,
+  },
+});
+
+// provide permissions to describe the user pool scoped to the ARN of 'myuserpool'
+postAuthFn.role?.attachInlinePolicy(new iam.Policy(this, 'userpool-policy', {
+  statements: [new iam.PolicyStatement({
+    actions: ['cognito-idp:DescribeUserPool'],
+    resources: [userpool.userPoolArn],
+  })],
+}));
+```
```

Comment:
This is 15 lines of code with only the last 5 lines of actual value. Anything we can do to reduce the amount of boilerplate?

Not a major comment.

Reply:
I could create a new fixture specifically for this snippet. would you prefer that here? - it's primarily this long for rosetta compatibility

Reply:
Either that or linking to the specific issue comment that has this workaround?

Reply:
created a fixture (didn't quite read your comment). we can also link to comment if it's helpful, but i figured at least the fixture will help users in other languages as well.